tests/behat/src/FixtureContext.php

```diff
@@ -0,0 +1,281 @@
+        assertNotNull($button, 'Close button not found');
+
+        $button->click();
+    }
+
+    /**
+     * @Given /^I click the "([^"]+)" gallery item$/
+     * @param $name
+     */
+    public function iClickTheGalleryItem($name)
+    {
+        /** @var DocumentElement $page */
+        $page = $this->getMainContext()->getSession()->getPage();
+        // Find by cell
+        $cell = $page->find(
+            'xpath',
+            "//div[contains(@class, 'gallery-item')]//div[contains(text(), '{$name}')]"
+        );
+
+        assertNotNull($cell, sprintf('Gallery item "%s" could not be found', $name));
+
+        $cell->click();
+    }
+
+    /**
+     * @Given /^I should( not? |\s*)see the "([^"]+)" campaign item$/
+     * @param $negate boolean
+     * @param $name string
+     */
+    public function iSeeTheCampaignItem($negate, $name)
+    {
+        $item = $this->getCampaignItem($name);
+        $shouldSee = !trim($negate);
+
+        if ($shouldSee) {
+            assertNotNull($item, sprintf('Item "%s" could not be found', $name));
+            assertTrue($item->isVisible(), sprintf('Item "%s" is not visible', $name));
+        } elseif ($item) {
+            assertFalse($item->isVisible(), sprintf('Item "%s" is visible', $name));
+        } else {
+            assertNull($item, sprintf('Item "%s" exists', $name));
+        }
+    }
+
+    /**
+     * @Given /^I select the "([^"]+)" campaign item$/
+     * @param $name
+     */
+    public function iSelectTheCampaignItem($name)
+    {
+        $item = $this->getCampaignItem($name);
+        assertNotNull($item, sprintf('Item "%s" could not be found', $name));
+        $item->click();
+    }
+
+    /**
+     * @Given /^a campaign "([^"]+)" with changes (".*)$/
+     * @param $id
+     * @param $data
+     */
+    public function stepCreateCampaignWithItems($id, $data)
+    {
+        $class = $this->convertTypeToClass(ChangeSet::class);
+        /* @var ChangeSet $campaign */
+        $campaign = $this->getFixtureFactory()->get($class, $id);
+        if (!$campaign) {
+            $campaign = $this->getFixtureFactory()->createObject($class, $id);
+        }
+
+        preg_match_all(
+            '/"(?<content_class>[^"]+)"\s*=\s*"(?<identifier>[^"]+)"/',
+            $data,
+            $matches
+        );
+        $itemMap = array_combine($matches['content_class'], $matches['identifier']);
+        foreach ($itemMap as $contentClass => $identifier) {
+            $class = $this->convertTypeToClass($contentClass);
+            $record = $this->getFixtureFactory()->get($class, $identifier);
+            if (!$record) {
+                $record = $this->getFixtureFactory()->createObject($class, $identifier);
+            }
+            $campaign->addObject($record);
+        }
+    }
+
+    /**
+     * @When /^I click the "([^"]+)" element$/
+     * @param $selector
+     */
+    public function iClickTheElement($selector)
```

Comment:
Duplicate of `SilverStripe\BehatExtension\Context\BasicContext::iClickInTheElement()`

Reply:
There was nothing I could find for clicking a specific element by selector. `iClickInTheElement()` looks for text in an element, which doesn't always work. For instance, a `<button />` that's just an empty tag with an icon in a `:before` pseudo class.

Reply:
If that's the case, I suggest to add it to that context. This is an extremely generic action, but on a very specific use case context.